StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift

```diff
@@ -401,12 +402,9 @@ extension EmbeddedPaymentElement: EmbeddedFormViewControllerDelegate {
             if let lastSelection, lastSelection != currentlySelectedType {
                 // Go back to the previous selection if there was one
                 embeddedPaymentMethodsView.resetSelectionToLastSelection()
-            } else {
-                // If there wasn't a previous selection, deselect if the selection isn't valid.
-                let isCurrentSelectionValid = embeddedFormViewController.selectedPaymentOption != nil
-                if !isCurrentSelectionValid {
-                    embeddedPaymentMethodsView.resetSelection()
-                }
+            } else if paymentOption != lastUpdatedPaymentOption {
+                // Reset selection if no previous selection exists and the closing form's payment option differs from last updated
```

Comment:
This comment is too much of a direct translation of the code IMO and not useful enough. Can we say what we're trying to do here at a higher level?

For example, the code above:
```
            if let lastSelection, lastSelection != currentlySelectedType {
                // Go back to the previous selection if there was one
```
This tells you what we're trying to accomplish and would hold true regardless of what the actual code looks like.

If the comment was a direct translation of the code it'd say something like:
`// Reset selection to last selection if there's a last selection and the last selection differs from the currently 
selected type.`

which isn't so helpful.

Reply:
updated!